src/main.rs

```diff
@@ -269,7 +272,8 @@ async fn main() {
         .nest_service("/attachments", ServeDir::new("attachments"))
         .with_state(state);
 
-    let addr = SocketAddr::from(([127, 0, 0, 1], args.port));
+    let server_details = format!("{}:{}", args.listen, args.port);
+    let addr: SocketAddr = server_details.parse().expect("Unable to parse socket address");
     println!("Server running on http://{}", addr);
 
     axum::serve(tokio::net::TcpListener::bind(&addr).await.unwrap(), app)
```

Comment:
Let's handle errors nicer while we're at it. Otherwise looks good to me, thank you for the contribution!

```suggestion
    match tokio::net::TcpListener::bind(&addr).await {
        Ok(listener) => {
            if let Err(e) = axum::serve(listener, app).await {
                println!("Server error: {}", e);
            }
        }
        Err(e) => {
            println!("Failed to bind to address {}: {}", addr, e);
        }
    }
```